File: architect_zenghui/src/main/java/com/east/architect_zenghui/architect_35_okhttp_rxjava_retrofit/MainActivity.java
package com.east.architect_zenghui.architect_35_okhttp_rxjava_retrofit;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.east.architect_zenghui.R;
import com.east.architect_zenghui.architect_35_okhttp_rxjava_retrofit.simple1.RetrofitClient;
import com.east.architect_zenghui.architect_35_okhttp_rxjava_retrofit.simple1.UserInfo;
import com.east.architect_zenghui.architect_35_okhttp_rxjava_retrofit.simple2.HttpCallback;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        RetrofitClient.getServiceApi().userLogin1("","")
                .enqueue(new HttpCallback<UserInfo>() {
                    @Override
                    public void onSucceed(UserInfo result) {

                    }

                    @Override
                    public void onError(String code, String msg) {

                    }
                });
    }
}

File: databinding/src/main/java/com/east/databinding/MainActivity.kt
package com.east.databinding

import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import com.east.databinding.bean.User
import com.east.databinding.databinding.UserBinding
import com.east.databinding.fragment.CustomerFragment

class MainActivity : AppCompatActivity(),CustomerFragment.OnFragmentInteractionListener {
    override fun onFragmentInteraction(uri: Uri) {

    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val userBinding = DataBindingUtil.setContentView<UserBinding>(this, R.layout.activity_main)
        userBinding.beanUserInfo = User("名称","密码",false)
        supportFragmentManager.beginTransaction()
            .replace(R.id.fl,CustomerFragment.newInstance("这是传递的用户名","这是密码"))
            .commitAllowingStateLoss()
    }
}
